src/Transports/MassTransit.EventHubIntegration/EventHubIntegration/EventHubProducer.cs

```diff
@@ -69,15 +81,14 @@ public async Task Produce<T>(object values, IPipe<EventHubSendContext<T>> pipe,
         public Task Produce<T>(IEnumerable<object> values, IPipe<EventHubSendContext<T>> pipe, CancellationToken cancellationToken)
             where T : class
         {
-            // TODO change this to use the proper header initialization, etc.
-
-            Task<InitializeContext<T>>[] messageTasks = values.Select(value => MessageInitializerCache<T>.Initialize(value, cancellationToken)).ToArray();
+            Task<SendTuple<T>>[] messageTasks = values.Select(value => MessageInitializerCache<T>.InitializeMessage(value, cancellationToken)).ToArray();
 
             async Task ProduceAsync()
```

Comment:
there isn't a non-async code path, so remove the complexity of it and just use async?